livekit-plugins/livekit-plugins-playht/livekit/plugins/playht/tts.py

```diff
@@ -0,0 +1,222 @@
+        stream = utils.audio.AudioByteStream(
+            sample_rate=self._opts.sample_rate, num_channels=1
+        )
+        self._mp3_decoder = utils.codecs.Mp3StreamDecoder()
+        request_id = utils.shortuuid()
+        segment_id = utils.shortuuid()
+        parent_path = os.path.dirname(os.path.abspath(__file__))
+        client = Client(self._opts.user_id, self._opts.api_key)
+        options = TTSOptions(
+            voice=self._opts.voice.id,
+            sample_rate=PLAYHT_TTS_SAMPLE_RATE,
+            format=Format.FORMAT_WAV,
+            speed=1
+        )
+
+        try:
+            url = "https://api.play.ht/api/v2/tts/stream"
+            headers = {
+                "accept": ACCEPT_HEADER[self._opts.encoding],
+                "content-type": "application/json",
+                "AUTHORIZATION": self._opts.api_key,
+                "X-USER-ID": self._opts.user_id
+            }
+            json_data = {
+                "text": "Hello, How are you?",
+                "output_format": self._opts.en,
+                "voice": "s3://voice-cloning-zero-shot/d9ff78ba-d016-47f6-b0ef-dd630f59414e/female-cs/manifest.json",
+            }
+            async with self._session.post(url=url, headers=headers, json=json_data) as resp:
+                if not resp.content_type.startswith("audio/"):
+                    content = await resp.text()
+                    logger.error("playHT returned non-audio data: %s", content)
+                    return
+                
+                encoding = _encoding_from_format(self._opts.encoding)
+                if encoding == "mp3":
+                    async for bytes_data, _ in resp.content.iter_chunks():
+                        for frame in self._mp3_decoder.decode_chunk(bytes_data):
+                            self._event_ch.send_nowait(
+                                tts.SynthesizedAudio(
+                                    request_id=request_id,
+                                    segment_id=segment_id,
+                                    frame=frame,
+                                )
+                            )
+                else:
+                    async for bytes_data, _ in resp.content.iter_chunks():
+                        for frame in stream.write(bytes_data):
+                            self._event_ch.send_nowait(
+                                tts.SynthesizedAudio(
+                                    request_id=request_id,
+                                    segment_id=segment_id,
+                                    frame=frame,
+                                )
+                            )
+
+                    for frame in stream.flush():
+                        self._event_ch.send_nowait(
+                            tts.SynthesizedAudio(
+                                request_id=request_id, segment_id=segment_id, frame=frame
+                            )
+                        )
+        except Exception as e:
```

Comment:
You can ignore the exception, is something is raised inside this task, it'll already be logged by the above decorator

`@utils.log_exceptions(logger=logger)`

Reply:
Yes, I remove it